sync3/lists.go

```diff
@@ -74,20 +74,32 @@ func (s *InternalRequestLists) SetRoom(r RoomConnMetadata, replacePreviousTimest
 			)
 		}
 
-		// Only bump this room in the room list if the update is of interest to the
-		// client.
-		if replacePreviousTimestamp {
-			r.LastInterestedEventTimestamp = r.LastMessageTimestamp
-		} else {
-			r.LastInterestedEventTimestamp = existing.LastInterestedEventTimestamp
+		// Interpret the timestamp map on r as the changes we should apply atop the
+		// existing timestamps.
+		bumpListsTo := r.LastInterestedEventTimestamps
+		r.LastInterestedEventTimestamps = make(map[string]uint64, len(s.lists))
+		for listKey := range s.lists {
+			newTs, bump := bumpListsTo[listKey]
```

Comment:
`bumpListsTo` is a really poor name, and I wrote a comment before realising that this is the new map and not the existing map. Can we call it `newTimestamps` or something?

Reply:
I can do this, but again I think part of the confusion is that we are passing in an entirely new RoomConnMetadata `r` as an argument to this function. We really want to pass in a diff to an existing RoomConnMetadata.

Reply:
11ba045